nvflare/app_opt/sklearn/data_loader.py

```diff
@@ -55,14 +55,15 @@ def load_data(data_path: str, require_header: bool = False):
 
 def load_data_for_range(data_path: str, start: int, end: int, require_header: bool = False):
     reader = get_pandas_reader(data_path)
+    header = 'infer'
+
+    if (hasattr(reader, "header") != (not require_header)):
+        header = None
 
     if hasattr(reader, "skiprows"):
         data_size = end - start
-        if hasattr(reader, "header") and not require_header:
-            data = reader(data_path, header=None, skiprows=start, nrows=data_size)
-        else:
-            data = reader(data_path, skiprows=start, nrows=data_size)
+        data = reader(data_path, header=header, skiprows=start, nrows=data_size)
     else:
-        data = reader(data_path).iloc[start:end]
+        data = reader(data_path, header=header).iloc[start:end]
```

Comment:
Thanks for noticing this! I think we can just change previous line 66 to 
data = reader(data_path, header=None).iloc[start:end]
Meaning if reader does not find header, then it should read with header=None.
(the above line 51 also need to add "header=None")
Keeping all other lines untouched.

Reply:
In this case, on line 66, wouldn't it set header=None for every case where the reader doesn't find a "skiprows" attribute? Even if there is a header?

Reply:
ah you are right, we need another "if hasattr(reader, "header") and not require_header: else:" for none "skiprow" cases. My point is mainly for not introducing an "arbitrary" header of "infer".

Reply:
and it should be 
if hasattr(reader, "header") and require_header:
            data = reader(data_path, skiprows=start, nrows=data_size)
else:
            data = reader(data_path, header=None, skiprows=start, nrows=data_size)

Reply:
I've implemented the changes. Changed the logic from "not require_header" to "require_header". I also kept the `.iloc` for cases where "skiprows" is not supported (specifically for read_json). I think it should be fine now

Reply:
Thanks! could you also update line 45-49 to combine with line 51 removing the redundancy? using our updated condition logic.